target/scripts/helpers/log.sh

```diff
@@ -1,5 +1,103 @@
 #! /bin/bash
 
+LOG_RESET='\e[0m'
+LOG_LGRAY='\e[37m'
+LOG_LBLUE='\e[94m'
+LOG_BLUE='\e[34m'
+LOG_LYELLOW='\e[93m'
+LOG_RED='\e[91m'
+
+# ### DMS Logging Functionality
+#
+# This function provides the logging for scripts used by DMS.
+# It adheres to the convention for log levels.
+# Valid values (in order of increasing verbosity) are: `error`,
+# `warn`, `info`, `debug` and `trace`. The default log level
+# is `info`.
+#
+# #### Arguments
+#
+# $1 :: the log level to log the message with
+# $2 :: the message
+#
+# #### Panics
+#
+# If the first argument is not set or invalid, an error
+# message is logged. Likewise when the second argument
+# is missing. Both failures will return with exit code '1'.
+function _log
+{
+  if [[ -z ${1+set} ]]
+  then
+    echo "Call to '_log' is missing a valid log level" >&2
+    return 1
+  fi
+
+  if [[ -z ${2+set} ]]
+  then
+    echo "Call to '_log' is missing a message to log" >&2
+    return 1
+  fi
+
+  local MESSAGE LEVEL_AS_INT
+  MESSAGE="${LOG_RESET}["
+
+  case "${LOG_LEVEL:-}" in
+    ( 'trace'  ) LEVEL_AS_INT=5 ;;
+    ( 'debug'  ) LEVEL_AS_INT=4 ;;
+    ( 'warn'   ) LEVEL_AS_INT=2 ;;
+    ( 'error'  ) LEVEL_AS_INT=1 ;;
+    ( *        ) LEVEL_AS_INT=3 ;;
+  esac
+
+  case "${1}" in
+    ( 'trace' )
+      [[ ${LEVEL_AS_INT} -ge 5 ]] || return 0
+      MESSAGE+="  ${LOG_LGRAY}TRACE  "
+      ;;
+
+    ( 'debug' )
+      [[ ${LEVEL_AS_INT} -ge 4 ]] || return 0
+      MESSAGE+="  ${LOG_LBLUE}DEBUG  "
+      ;;
+
+    ( 'info' )
+      [[ ${LEVEL_AS_INT} -ge 3 ]] || return 0
+      MESSAGE+="   ${LOG_BLUE}INF   "
```

Comment:
1. `_notify` uses `INF` as well :)
2. This would break my much-loved symmetry

I think you have a small misconception about the message format I chose (seeing https://github.com/docker-mailserver/docker-mailserver/pull/2493#discussion_r830541017 as well): With this PR, log looks like this:

``` TXT
[  TRACE  ]  Some very verbose trace message
[  DEBUG  ]  Some verbose debug message
[   INF   ]  Some info message
[ WARNING ]  Some warning message
[  ERROR  ]  Some error message
```

I can change it to `INFO` if this is desired by a majority (or if we decide it at a later point in time), but I'd like to keep it the way it is (at least for now) :) This also requires the spaces around the strings (these are not there by accident) :D